### DISCUSSION TITLE
schema.enable_dynamic_field

### DISCUSSION THREAD

**Discussion by reenee** [milvus-io/milvus#28358] [Q&A and General discussion]
Hi Team, is there a way to enable schema.enable_dynamic_field for an existing collection using python sdk or rest API?

Above snippet  doesnt work for me
`schema = collection.schema
print("Collection schema dynamic field enabled?  %s",schema.enable_dynamic_field)
if not schema.enable_dynamic_field:
    schema.enable_dynamic_field = True
    # Update schema
    collection.schema = schema
    collection.load()
    print("Collection schema dynamic field enabled!!")`

**Comment by xiaofan-luan** [comment]
you have to set dynamic field  enabled when you create collection.
like this:
# 1. define fields
fields = [
    FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=True),
    FieldSchema(name="title", dtype=DataType.VARCHAR, max_length=512),
    FieldSchema(name="title_vector", dtype=DataType.FLOAT_VECTOR, dim=768)
]
# 2. enable dynamic schema in schema definition
schema = CollectionSchema(
        fields, 
        "The schema for a medium news collection", 
        enable_dynamic_field=True
)

# 3. reference the schema in a collection
collection = Collection("medium_articles_with_dynamic", schema)

**Comment by xiaofan-luan** [comment]
https://milvus.io/docs/dynamic_schema.md#:~:text=Create%20collection%20with%20dynamic%20schema,treated%20as%20pre%2Ddefined%20fields.

**Reply by reenee** [reply]
Hi , is there an option to update the **existing collection**  and toggle **enable_dynamic_field** property?

**Comment by xiaofan-luan** [comment]
So far the answer is no. Milvus doens't support schema change.
But we are actively working on it
